Zend/tests/attributes/deprecated/class_constants/deprecated_constant_as_message_001.phpt

```diff
@@ -0,0 +1,26 @@
+--TEST--
+#[\Deprecated]: Using the value of a deprecated class constant as the deprecation message.
+--FILE--
+<?php
+
+class Clazz {
+	#[\Deprecated(self::TEST)]
+	public const TEST = "from itself";
+
+	#[\Deprecated]
+	public const TEST2 = "from another";
+
+	#[\Deprecated(self::TEST2)]
+	public const TEST3 = 1;
+}
+
+Clazz::TEST;
+Clazz::TEST3;
+
+?>
+--EXPECTF--
+Deprecated: Constant Clazz::TEST is deprecated, from itself in %s on line %d
```

Comment:
Technically it would need to emit infinite deprecation notices then. In any case this is effectively pre-existing behavior, as I am not doing anything special here. This test primarily exists to verify that no memory-unsafety happens for this kind of edge case. I don't expect it to be relevant in the real world.

Reply:
Fair point. I'd still be interested in knowing _why_ this doesn't trigger twice though, and if it's intentional or just a happy accident (that may break in the future).